### DISCUSSION TITLE
ook file - routinely clear contents

### DISCUSSION THREAD

**Discussion by frevoli** [merbanan/rtl_433#2874] [Q&A]
Sometimes, whilst long term soak testing, my output ook file grows too large. 

Is there a clean way to clear its contents every 'x' days, or when a specific file size is reached?  

I have tried to do this using Python, which works, but if it happens to clear the file whilst rtl_433 is writing to it then rtl_433 crashes. 

I considered maybe stopping my rtl_433 process, deleting the ook file, then restarting rtl_433, but this feels unnecessarily disjointed.

Anyone have any ideas as to how I might achieve this cleanly?

Many thanks!

**Comment by zuckschwerdt** [comment]
Interesting. It should be possible to enable this and e.g. automate it with logrotate. Works in this way:
You rename the current output file and signal the running program (`killall -HUP rtl_433`).
The program then checks and reopens all renamed files.

We'll just have to implement the signal/reopen part.

**Reply by zuckschwerdt** [reply]
There is a draft PR in #2876 to allow this for dumpers, i.e. `-w myfile.ook` -- please test :)

**Reply by frevoli** [reply]
Thank you. I shall!

**Reply by frevoli** [reply]
So it turns out that ook filesize was a bit of a red herring! 

Closer inspection suggests that the reason my rtl-433 instances crash after a period is in fact due to the error...
"Async read stalled, exiting!"

How do I prevent this from occurring?

Thanks again.

**Reply by zuckschwerdt** [reply]
That's your RTL-SDR dongle erroring. Nothing we can do about that. But there is the `-D restart` option which might recover the SDR on failure.

**Reply by frevoli** [reply]
Interesting.  I run two SDRs simultaneously, 433MHz & 868MHz.
The 868 generally fails first.

I will try the -D restart option as suggested.

Thank you.

**Reply by zuckschwerdt** [reply]
There might be a reason: thermal problems manifest with higher sample rates and 868 has 1M sample rate, 433 only 250k.
If those are the cheapest RTL-SDR then maybe glue a heatsink on.

**Reply by frevoli** [reply]
I'll experiment with heatsinks in addition to the restart flag

**Reply by frevoli** [reply]
Seems ok so far. I'll keep you updated of the outcome.
Incidentally, you mention 868 uses 1M SR, and 433 uses 250K SR.
I use 1M for both. It works just as well.
